File: leet/0720_pre/leet_561.py
#given an array of 2n ints, group these integers into n pairs which makes sum of pair_i for all i 
#from 1 to n as large as possible
class Solution(object):
    def arrayPairSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        leng = len(nums)
        nums.sort()
        res = 0
        for i in range(0, leng, 2):
            res += nums[i]
        return res

if __name__ == '__main__':
    solu = Solution()
    print solu.arrayPairSum([1,4,3,2])

File: leet/0720_pre/leet_561.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int arrayPairSum(vector<int>& nums) {
        int len = nums.size();
        //int half_len = len / 2;
        sort(nums.begin(), nums.end());
        int sum = 0;
        for(int i = 0; i < len; i = i+2)
        {
            sum += nums.at(i);
        }
        return sum;
    } 
};

int main()
{
    int a[] = {1,4,3,2};
    vector<int> arr(a, a + sizeof(a)/sizeof(a[0]));
    Solution solu;
    cout << solu.arrayPairSum(arr) << endl;
    return 0;
}
